parser/parser_system1.c

```diff
@@ -174,16 +174,18 @@ static int parse_bsp_drivers(struct pv_state *s, char *v, int len)
 
 	// take null terminate copy of item to parse
 	buf = calloc(len + 1, sizeof(char));
-	buf = memcpy(buf, v, len);
 
 	if (!buf)
 		return 0;
 
+	buf = memcpy(buf, v, len);
+
 	jsmnutil_parse_json(buf, &tokv, &tokc);
 
 	keys = jsmnutil_get_object_keys(buf, tokv);
 	if (!keys) {
 		pv_log(ERROR, "drivers list cannot be parsed");
+		free(buf);
```

Comment:
github doesn't let me coment in line 207, but we would miss a "free(buf);" before "return 0;" I suppose